### PR TITLE
Issue #7102 - WebClient should have a mode that is resilient to bad media/content types

### PR DESCRIPTION
**Pull request by Tomas-Kraus**
fixes issue #7102
backport of issue #6774 fix

### CODE REVIEW

File: tests/integration/webclient/src/test/java/io/helidon/tests/integration/webclient/HeadersTest.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    }
+
+    // Verify that invalid content type causes an exception when being parsed
+    @Test
+    void testInvalidContentType() {
+        try {
+            client.get()
+                    .path("/invalidContentType")
+                    .request()
+                    .await();
```
Comment (Verdent) [inline tests/integration/webclient/src/test/java/io/helidon/tests/integration/webclient/HeadersTest.java:85]: Please set timeout to the await method.

File: tests/integration/webclient/src/test/java/io/helidon/tests/integration/webclient/HeadersTest.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    }
+
+    // Verify that "text" content type causes an exception when being parsed in strict mode
+    @Test
+    void testInvalidTextContentTypeStrict() {
+        try {
+        client.get()
+                .path("/invalidTextContentType")
+                .request()
+                .await();
```
Comment (Verdent) [inline tests/integration/webclient/src/test/java/io/helidon/tests/integration/webclient/HeadersTest.java:103]: add timeout please.

File: tests/integration/webclient/src/test/java/io/helidon/tests/integration/webclient/HeadersTest.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        WebClient client = WebClient.builder()
+                .baseUri("http://localhost:" + server.port())
+                .validateHeaders(false)
+                .keepAlive(true)
+                .mediaTypeParserRelaxed(true)
+                .build();
+        WebClientResponse response = client.get()
+                .path("/invalidTextContentType")
+                .request()
+                .await();
```
Comment (Verdent) [inline tests/integration/webclient/src/test/java/io/helidon/tests/integration/webclient/HeadersTest.java:125]: add timeout please.

File: tests/integration/webclient/src/test/java/io/helidon/tests/integration/webclient/HeadersTest.java [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        Optional<MediaType> maybeContentType = response.headers().contentType();
+        assertThat(maybeContentType.isPresent(), is(true));
+        assertThat(maybeContentType.get().toString(), is(RELAXED_CONTENT_TYPE_TEXT));
+    }
+
+    private static final String INVALID_CONTENT_TYPE_VALUE = "invalid header value";
```
Comment (Verdent) [inline tests/integration/webclient/src/test/java/io/helidon/tests/integration/webclient/HeadersTest.java:131]: constants should be at the top of the class.

File: tests/integration/webclient/src/test/java/io/helidon/tests/integration/webclient/HeadersTest.java [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+    // HTTP service with invalid Content-Type
+    private static void invalidContentType(ServerRequest request, ServerResponse response) {
```
Comment (Verdent) [inline tests/integration/webclient/src/test/java/io/helidon/tests/integration/webclient/HeadersTest.java:134]: Move also these static methods under the beforeAll method please. I could not find them when trying to figure out what the endpoint does.

File: tests/integration/webclient/src/test/java/io/helidon/tests/integration/webclient/HeadersTest.java [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+            assertThat(cause, is(instanceOf(IllegalArgumentException.class)));
+            assertThat(cause.getMessage(), containsString(INVALID_CONTENT_TYPE_VALUE));
+        }
+    }
+
+    // Verify that "text" content type causes an exception when being parsed in strict mode
+    @Test
+    void testInvalidTextContentTypeStrict() {
+        try {
+        client.get()
```
Comment (Verdent) [inline tests/integration/webclient/src/test/java/io/helidon/tests/integration/webclient/HeadersTest.java:100]: I think this formatting should be padded.

**Review by Verdent — CHANGES_REQUESTED** [review]
I have a 2 questions. :-)
1. Why have you decided to drop ParserMode enum, which was used in the 4.x implementation?
2. I think I have not seen it also in the 4.x version, but should not these relaxed formats of the content types be also configurable? I mean if user wants to add their own.

**Comment by Tomas-Kraus** [comment]
>     1. Why have you decided to drop ParserMode enum, which was used in the 4.x implementation?
MediaType in 4.x is rewritten. It's parse method is not public so changing prototype did not cause any issues.
MediaType parse method in 3.x is public static and as I understand it properly, it shall match current prototype to be used as method reference. That's the reason why I made it without ParserMode enum and mode is choosen by the method being called.

>     2. I think I have not seen it also in the 4.x version, but should not these relaxed formats of the content types be also configurable? I mean if user wants to add their own.
4.x version has hard-coded relaxed format as well. I made it the way jeff requested. This issue is just backport, not any additional enhancement to be done in 3.x only.

**Review by tomas-langer — APPROVED** [review]

**Review by Verdent — APPROVED** [review]

**Pull request merged**